### PR TITLE
feat(vehicle/lights): update vehicle lights natives

### PR DESCRIPTION
**Pull request by ahcenezdh**
* Update `SET_VEHICLE_HAS_UNBREAKABLE_LIGHTS`
* Update `SET_VEHICLE_HEADLIGHT_SHADOWS`
* Update `SET_VEHICLE_LIGHTS`

### CODE REVIEW

**Comment by ahcenezdh** [comment]
Supersedes #1187

File: VEHICLE/SetVehicleHeadlightShadows.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+## Examples
+```lua
+local vehicle = GetVehiclePedIsIn(PlayerPedId(), false)
+if not vehicle then return end
+
+-- Set the vehicle headlight shadows to cast static shadows (always on)
+SetVehicleHeadlightShadows(vehicle, 2)
+```
+
+### JavaScript Example
```
Comment (AvarianKnight) [inline VEHICLE/SetVehicleHeadlightShadows.md:40]:
```suggestion

```

File: VEHICLE/SetVehicleHeadlightShadows.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+```js
+const vehicle = GetVehiclePedIsIn(PlayerPedId(), false);
+if (!vehicle) return;
+
+// Set the vehicle headlight shadows to cast static shadows (always on)
+SetVehicleHeadlightShadows(vehicle, 2);
+```
+
+### C# Example
```
Comment (AvarianKnight) [inline VEHICLE/SetVehicleHeadlightShadows.md:49]:
```suggestion

```

File: VEHICLE/SetVehicleHeadlightShadows.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+```
+
+Sets the vehicle headlight shadow flags.
+
+```
+NativeDB Introduced: v323
+```
+
+```c
+enum vehicleHeadlightShadowFlags {
```
Comment (AvarianKnight) [inline VEHICLE/SetVehicleHeadlightShadows.md:19]:
```suggestion
enum eVehicleHeadlightShadowFlags {
```

We should keep enum naming consistent with other natives.

File: VEHICLE/SetVehicleHeadlightShadows.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    NO_HEADLIGHT_SHADOWS            = 0, // Default (Lights can be toggled between off, normal and high beams)
+    HEADLIGHTS_CAST_DYNAMIC_SHADOWS = 1, // Lights Disabled (Lights are fully disabled, cannot be toggled)
+    HEADLIGHTS_CAST_STATIC_SHADOWS  = 2, // Always On (Lights can be toggled between normal and high beams)
+    HEADLIGHTS_CAST_FULL_SHADOWS    = 3 
+};
+```
+
+## Parameters
+* **vehicle**: The target vehicle.
+* **flag**: A value from `vehicleHeadlightShadowFlags` representing the desired headlight shadow flag.
```
Comment (AvarianKnight) [inline VEHICLE/SetVehicleHeadlightShadows.md:29]:
```suggestion
* **flag**: A value from `eVehicleHeadlightShadowFlags` representing the desired headlight shadow flag.

```

File: VEHICLE/SetVehicleLights.md [hunk 5]
```diff
@@ -1,26 +1,56 @@
...
+
+```c
+enum vehicleLightSetting {
+    NO_VEHICLE_LIGHT_OVERRIDE = 0, // Normal light behavior. Lights cycle through off, then low beams, then high beams. Note: It's affected by day or night; high beams don't exist in daytime.
+    FORCE_VEHICLE_LIGHTS_OFF  = 1, // Vehicle doesn't have lights, always off.
+    FORCE_VEHICLE_LIGHTS_ON   = 2, // Vehicle has always-on lights. During day: Cycles between low beams and high beams. At night: Cycles between low beams, low beams, and high beams.
+    SET_VEHICLE_LIGHTS_ON     = 3, // Sets vehicle lights on. Behaves like normal lights (same as 0).
+    SET_VEHICLE_LIGHTS_OFF    = 4  // Sets vehicle lights off. Behaves like normal lights (same as 0).
+};
+```
```
Comment (AvarianKnight) [inline VEHICLE/SetVehicleLights.md:25]:
````suggestion
```c
enum eVehicleLightSetting {
    // Normal light behavior. Lights cycle through off, then low beams, then high beams.
    // Note: It's affected by day or night; high beams don't exist in daytime.
    NO_VEHICLE_LIGHT_OVERRIDE = 0,
    // Vehicle doesn't have lights, always off.
    FORCE_VEHICLE_LIGHTS_OFF  = 1, 
    // Vehicle has always-on lights.
    // During day: Cycles between low beams and high beams. 
    // At night: Cycles between low beams, low beams, and high beams.
    FORCE_VEHICLE_LIGHTS_ON   = 2,
    // Sets vehicle lights on. Behaves like normal lights (same as 0).
    SET_VEHICLE_LIGHTS_ON     = 3,
    // Sets vehicle lights off. Behaves like normal lights (same as 0).
    SET_VEHICLE_LIGHTS_OFF    = 4 
};
```

````

Keep enum naming consistent & try to make comments easier to read

File: VEHICLE/SetVehicleLights.md [hunk 6]
```diff
@@ -1,26 +1,56 @@
...
+
+## Parameters
+* **vehicle**: The target vehicle.
+* **state**: A value from `vehicleLightSetting` representing the desired light setting.
```
Comment (AvarianKnight) [inline VEHICLE/SetVehicleLights.md:29]:
```suggestion
* **state**: A value from `eVehicleLightSetting` representing the desired light setting.

```

**Review by AvarianKnight — CHANGES_REQUESTED** [review]

File: VEHICLE/SetVehicleHeadlightShadows.md (hunk 1 above)
Comment (ahcenezdh) [reply VEHICLE/SetVehicleHeadlightShadows.md:40]: Sorry was doing some test

**Review by AvarianKnight — CHANGES_REQUESTED** [review]

**Pull request merged**